src/components/NihLinkWarning.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect } from 'react'
+import { div } from 'react-hyperscript-helpers'
+import { clearNotification, notify } from 'src/components/Notifications'
+import { authStore } from 'src/libs/auth'
+import * as Utils from 'src/libs/utils'
+import * as Profile from 'src/pages/Profile'
+
+
+export const NihLinkWarning = () => {
+  const { nihStatus } = Utils.useAtom(authStore)
+
+  useEffect(() => {
+    const notificationId = 'nih-link-warning'
+
+    if (nihStatus) {
+      const { linkedNihUsername, linkExpireTime } = nihStatus
+      const expireTime = linkExpireTime * 1000
+      const expired = expireTime < Date.now()
+      const expiringSoon = !expired && expireTime < (Date.now() + (1000 * 60 * 60 * 24))
```

Comment:
I'd do some inlining of these, since they're simple and only used once, but that may just be stylistic.